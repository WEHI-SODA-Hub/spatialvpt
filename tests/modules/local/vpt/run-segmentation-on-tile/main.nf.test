nextflow_process {

    name "Test Process RUN_SEGMENTATION_ON_TILE"
    script "modules/local/vpt/run-segmentation-on-tile/main.nf"
    process "RUN_SEGMENTATION_ON_TILE"

    test("Should run without failures") {

        when {
            params {
                outdir = "results"
            }
            process {
                """
                input[0] = Channel.of([
                    [ id: 'small_set' ],
                    [ file('${projectDir}/data/segmentation_specification.json', checkIfExists: true) ],
                    [ file('${projectDir}/data/202305010900_U2OS_small_set_VMSC00000/region_0/images', checkIfExists: true) ],
                    [ file('${projectDir}/data/example_analysis_algorithm/cellpose_default_1_ZLevel.json', checkIfExists: true) ],
                    0
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
