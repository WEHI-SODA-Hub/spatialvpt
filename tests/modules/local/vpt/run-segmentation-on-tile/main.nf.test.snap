{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "small_set"
                        },
                        "cell_0.parquet:md5,a7803f7f513ecb81c9ccf110d2958d1b"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b163ee9605979e1b9e8964bba3f7a88f"
                ],
                "segmented_tile": [
                    [
                        {
                            "id": "small_set"
                        },
                        "cell_0.parquet:md5,a7803f7f513ecb81c9ccf110d2958d1b"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b163ee9605979e1b9e8964bba3f7a88f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-08-03T12:02:54.459174138"
    }
}